File: src/02.string/03.count/04.242.ts
export function isAnagram(s: string, t: string): boolean {
    const sLen = s.length;
    const tLen = t.length;
    if (sLen !== tLen) {
        return false;
    }

    const obj: { [key: string]: number } = {};

    for (let i = 0; i < sLen; i++) {
        obj[s[i]] = obj[s[i]] ? obj[s[i]] + 1 : 1;
    }

    for (let i = 0; i < sLen; i++) {
        if (obj[t[i]] === undefined) {
            obj[t[i]] = 0;
        }
        obj[t[i]]--;
        if (obj[t[i]] < 0) {
            return false;
        }
    }

    return true;
}

File: src/02.string/03.count/04.242.go
package count

func IsAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	obj := make(map[rune]int)

	for _, v := range s {
		obj[v]++
	}

	for _, v := range t {
		obj[v]--
		if obj[v] < 0 {
			return false
		}
	}

	return true
}
